src/Service/Startup.cs

```diff
@@ -70,6 +75,11 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton(fileSystem);
             services.AddSingleton(configProvider);
             services.AddSingleton(configLoader);
+
+            // Register a TelemetryClient instance and custom ITelemetryInitializer implementation with the dependency injection
+            services.AddSingleton<TelemetryClient>();
```

Comment:
Why not use `services.AddApplicationInsightsTelemetry()`? That automatically registers `telemetryclient` for you and no need for static instance.

Reply:
Thanks, updated it accordingly.

Reply:
for static field, i created it to use telemetryClient in Program.cs when using constructing Provider `ApplicationInsightsLoggerProvider` for ILogger